### ISSUE TITLE
RCA: all Pixel 4a devices were found in recovery mode

### ISSUE THREAD

**Issue by greenEkatherine** [dotnet/xharness#619]
Since May 20th and following days all new Pixel 4a devices (DNCENGWIN046 and above) lost connection and were found in recovery mode https://github.com/dotnet/core-eng/issues/13183
Now they are recovered and back online but we need to find a root cause and prevent it in the future.

**Comment by greenEkatherine** [comment]
I see a pattern on many machines after the last passed test, other tests started failing to install apps with the same exception:

```
[14:49:42] fail: Error:
                 Exit code: 1
                 Standard Output:
                 Performing Streamed Install
                 
                 Standard Error:
                 adb: failed to install C:\h\w\B67409AF\w\C1BE0A3D\e\System.CodeDom.Tests.apk: 
                 Exception occurred while executing 'install':
                 java.lang.NullPointerException: Attempt to invoke virtual method 'void android.content.pm.PackageManagerInternal.freeStorage(java.lang.String, long, int)' on a null object reference
                       at com.android.server.StorageManagerService.allocateBytes(StorageManagerService.java:3901)
                       at android.os.storage.StorageManager.allocateBytes(StorageManager.java:2278)
                       at android.os.storage.StorageManager.allocateBytes(StorageManager.java:2335)
                       at com.android.server.pm.PackageI
```

I didn’t see this exception before, when I was fixing PACKAGE_NOT_INSTALLED error, it usually was just timeout.

**Comment by greenEkatherine** [comment]
It's interesting that DNCENGWIN-022 is also a device with API 30. I'm taking it offline to investigate while the device there is still working
![image](https://user-images.githubusercontent.com/19972657/120655665-32626080-c483-11eb-8dbe-bf6c098ec35d.png)

**Comment by greenEkatherine** [comment]
I did analysis of the device connected to DNCENGWIN-022 and it really helps. There were many installations failed and overall it looked like that device goes soon into recovery mode as it was with all other devices with API 30.

Storage looks good - only 15%, so it wasn't a clean up issue:
![image](https://user-images.githubusercontent.com/19972657/121192606-6667c800-c86d-11eb-97c7-72e51b2d6a40.png)

The only suspicious thing I found was RAM consumption:
![image](https://user-images.githubusercontent.com/19972657/121192820-9911c080-c86d-11eb-95e6-d9cb59155fb5.png)

It's about 10x more than on healthy devices. I looked into running processes and found that one process using almost 2GB:
![image](https://user-images.githubusercontent.com/19972657/121193268-fd348480-c86d-11eb-89d6-f9623761d2ae.png)

`com.android.vending` is related to the installer. The process running for almost 11 days now, my suggestion is that it's hanging up and increasing memory consumption after each installation attempt failing with timeout.

I'm going to raise PR in xharness to prevent it. In case when it failed during apk installation we should reboot a device and I consider to increase timeout up to 10 minutes for the _second_ attempt.

Also, I return back online DNCENGWIN-022 - it's back to normal after the device's rebooting:
![image](https://user-images.githubusercontent.com/19972657/121212526-bbabd580-c87d-11eb-9b1a-0916a728d5d3.png)
